src/components/InngestCommHandler.ts

```diff
@@ -438,7 +439,7 @@ export class InngestCommHandler<H extends Handler, TransformedRes> {
         if (stepRes.status === 500 || stepRes.status === 400) {
           return {
             status: stepRes.status,
-            body: stepRes.error || "",
```

Comment:
This is the offending line: `JSON.marshal(stepRes.error)` will always return `{}` due to how errors are transformed into JSON.